src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx

```diff
@@ -357,16 +377,20 @@ const NodeDetailComponent = ({
                 <Switch>
                     <Route path={`${path}/${NodeDetailTab.MANIFEST}`}>
                         <ManifestComponent
+                            key={getComponentKeyFromParams()}
```

Comment:
#### The need for this ? (For personal & future reference)

When a component's position in the dom tree doesn't change (which in this case was happening) the components state (not referring to useState state but the internal react state of a component) is preserved i.e The component never dismounts. 

To overcome this and make the necessary network calls on resource change from inside Manifest.component, the useEffect making the network calls (using Promise.all) had params.nodeType ... as dependencies. It was working perfectly fine.

When the state uplifting using a useRef was introduced (manifestViewRef), the changes to the ref are stored only when something changes (using useAfterComponentMount). The id was also being put in the dep list of useAfterComponentMount. Since the component never dismounted, this effect will run on resource change. Due to the asynchronous nature of useEffects we cannot ascertain if the id in the ref is the id of the old resources ref data.

Thus for simplicity we ought to just treat resource change as an event that necessitates treating the manifestComponent as a new entity. This is achieved using the key. Thus the component will now dismount and remount.